src/core/DataFile.cpp

```diff
@@ -49,6 +49,15 @@
 static void findIds(const QDomElement& elem, QList<jo_id_t>& idList);
 
 
+// Vector with all the upgrade methods
+const std::vector<void(DataFile::*)()> DataFile::m_upgradeMethods = {
+	&DataFile::upgrade_0_2_1_20070501, &DataFile::upgrade_0_2_1_20070508, &DataFile::upgrade_0_3_0_rc2,
+	&DataFile::upgrade_0_3_0, &DataFile::upgrade_0_4_0_20080104, &DataFile::upgrade_0_4_0_20080118,
+	&DataFile::upgrade_0_4_0_20080129, &DataFile::upgrade_0_4_0_20080409, &DataFile::upgrade_0_4_0_20080607,
+	&DataFile::upgrade_0_4_0_20080622, &DataFile::upgrade_0_4_0_beta1, &DataFile::upgrade_0_4_0_rc2,
+	&DataFile::upgrade_1_0_99, &DataFile::upgrade_1_1_0, &DataFile::upgrade_1_1_91, &DataFile::upgrade_1_2_0_rc3,
+	&DataFile::upgrade_1_3_0
```

Comment:
I feel like this should be formatted as a long list to be more readable.
```suggestion
	&DataFile::upgrade_0_2_1_20070501,
	&DataFile::upgrade_0_2_1_20070508,
	&DataFile::upgrade_0_3_0_rc2,
	&DataFile::upgrade_0_3_0,
	&DataFile::upgrade_0_4_0_20080104,
	&DataFile::upgrade_0_4_0_20080118,
	&DataFile::upgrade_0_4_0_20080129,
	&DataFile::upgrade_0_4_0_20080409,
	&DataFile::upgrade_0_4_0_20080607,
	&DataFile::upgrade_0_4_0_20080622,
	&DataFile::upgrade_0_4_0_beta1,
	&DataFile::upgrade_0_4_0_rc2,
	&DataFile::upgrade_1_0_99,
	&DataFile::upgrade_1_1_0,
	&DataFile::upgrade_1_1_91,
	&DataFile::upgrade_1_2_0_rc3,
	&DataFile::upgrade_1_3_0
```

Reply:
True! I'll update it

Reply:
Could we at least list, say, two per row? We can leave this as is until we're sure that the function names are final, but personally I'd like for it to be more compact. I think the comment, variable name, and the way the vector is used later make it pretty clear that it just lists all upgrade routines in order. I'm not sure it's particularly necessary to actually read the whole list for the sake of comprehending the code. Most modifications to it should just be appending another method at the end.

That said, if we want it to be a bit neater despite the compactness, perhaps we could align the columns with spaces. So

```
std::vector ... {
[Tab]upgradeRoutine,[Spaces       ] upgradeRoutine3,
[Tab]anotherUpgradeRoutine,[Spaces] upgradeRoutine4
}
```